lib/services/cups.pm

```diff
@@ -18,31 +18,47 @@ use utils;
 use strict;
 use warnings;
 
+my $service_type = 'Systemd';
+my $service_stage;
+
 sub install_service {
     zypper_call("in cups", exitcode => [0, 102, 103]);
 }
 
 sub config_service {
-    script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
-    validate_script_output 'cupsd -t', sub { m/is OK/ };
+    if ($service_type eq 'Systemd') {
+        script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
+        validate_script_output 'cupsd -t', sub { m/is OK/ };
+    } else {
+        # Allow new printers to be added using device URIs "file:/filename"
+        script_run 'echo FileDevice Yes >> /etc/cups/cupsd.conf';
+        assert_script_run 'cupsd';
+    }
 }
 
 sub enable_service {
-    systemctl 'enable cups.service';
+    common_service_action 'cups', $service_type, 'enable';
 }
 
 sub start_service {
-    systemctl 'start cups.service';
+    common_service_action 'cups', $service_type, 'start';
 }
 
 # check service is running and enabled
 sub check_service {
-    systemctl 'is-enabled cups.service';
-    systemctl 'is-active cups';
+    common_service_action 'cups', $service_type, 'is-enabled';
+    common_service_action 'cups', $service_type, 'is-active';
 }
 
 # check cups function
 sub check_function {
+    # if we migrated from sle11sp4, we need change the configure file
+    if ($service_stage eq 'after' && check_var('ORIGIN_SYSTEM_VERSION', '11-SP4')) {
+        script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
```

Comment:
Why need update the config file for cups when migration from SLES11SP4? On my understanding, this should be done during migration automatically.

Reply:
Because on 15sp3 we have different configure files againt 11sp4. I don't know why migration can't handle that.
just want to make it works on 15sp3.

Reply:
If it is not work automatically after migration from SLES11SP4, I think it is a bug and we can do something to make it work but should be a workaround, WDYT?

Reply:
OK

Reply:
bug filed and changed code and rerun